File: 3606-MinimumElementAfterReplacementWithDigitSum/3606-MinimumElementAfterReplacementWithDigitSum.js
// Last updated: 3/6/2026, 2:08:12 PM
/**
 * @param {number[]} nums
 * @return {number}
 */
var minElement = function(nums) {
    let res = [];
    for(let i=0;i<nums.length;i++){
        let sum = 0;
        let temp = nums[i];
        while(temp>0){
            let digit = temp % 10;
            sum = sum + digit;
            temp = Math.floor(temp / 10);
        }
        res.push(sum);
    }
    return Math.min(...res);
};

File: 3606-MinimumElementAfterReplacementWithDigitSum/3606-MinimumElementAfterReplacementWithDigitSum.cpp
// Last updated: 7/17/2026, 4:23:46 PM
class Solution {
public:
    int minElement(vector<int>& nums) {
        vector<int> res;

        for (int i = 0; i < nums.size(); i++) {
            int sum = 0;
            int temp = nums[i];

            while (temp > 0) {
                int digit = temp % 10;
                sum += digit;
                temp = temp / 10;
            }

            res.push_back(sum);
        }

        int ans = res[0];
        for (int i = 1; i < res.size(); i++) {
            ans = min(ans, res[i]);
        }

        return ans;
    }
};